test/conftest.py

```diff
@@ -734,3 +777,157 @@ def sampleset_ref(clients: dict[str, Any], ws_id: int, reads_refs: list[str]) ->
+    set_api_client: SetAPI,
+    context: dict[str, str | list[Any]],
+    ws_id: int,
+) -> dict[str, Any]:
+    return save_set(EXPRESSION, set_api_client, context, expression_refs, ws_id)
+
+
+@pytest.fixture(scope="session")
+def feature_set_set(
+    feature_set_refs: list[str],
+    set_api_client: SetAPI,
+    context: dict[str, str | list[Any]],
+    ws_id: int,
+) -> dict[str, Any]:
+    return save_set(FEATURE_SET, set_api_client, context, feature_set_refs, ws_id)
+
+
+@pytest.fixture(scope="session")
+def empty_feature_set_set(
+    set_api_client: SetAPI, context: dict[str, str | list[Any]], ws_id: int
+) -> dict[str, Any]:
+    return save_empty_set(FEATURE_SET, set_api_client, context, ws_id)
+
+
+@pytest.fixture(scope="session")
+def genome_set(
+    genome_refs: list[str],
+    set_api_client: SetAPI,
+    context: dict[str, str | list[Any]],
+    ws_id: int,
+) -> dict[str, Any]:
+    return save_set(GENOME, set_api_client, context, genome_refs, ws_id)
+
+
+@pytest.fixture(scope="session")
+def empty_genome_set(
+    set_api_client: SetAPI, context: dict[str, str | list[Any]], ws_id: int
+) -> dict[str, Any]:
+    return save_empty_set(GENOME, set_api_client, context, ws_id)
+
+
+@pytest.fixture(scope="session")
+def kbase_search_genome_set(
+    genome_refs: list[str],
+    set_api_client: SetAPI,
+    context: dict[str, str | list],
+    ws_id: int,
+) -> dict[str, Any]:
+    """Set of KBaseSearch genomes."""
+    return save_kbase_search_set(set_api_client, context, genome_refs, ws_id)
+
+
+@pytest.fixture(scope="session")
+def reads_set(
+    reads_refs: list[str],
+    set_api_client: SetAPI,
+    context: dict[str, str | list[Any]],
+    ws_id: int,
+) -> dict[str, Any]:
+    return save_set(READS, set_api_client, context, reads_refs, ws_id)
+
+
+@pytest.fixture(scope="session")
+def empty_reads_set(
+    set_api_client: SetAPI, context: dict[str, str | list[Any]], ws_id: int
+) -> dict[str, Any]:
+    return save_empty_set(READS, set_api_client, context, ws_id)
+
+
+@pytest.fixture(scope="session")
+def reads_alignment_set(
+    alignment_refs: list[str],
+    set_api_client: SetAPI,
+    context: dict[str, str | list[Any]],
+    ws_id: int,
+) -> dict[str, Any]:
+    return save_set(READS_ALIGNMENT, set_api_client, context, alignment_refs, ws_id)
+
+
+@pytest.fixture(scope="session")
+def random_fixture(request: FixtureRequest) -> dict[str, Any]:
+    """Pick a random fixture to use for testing."""
+    fixtures = [
+        "assembly_set",
+        "differential_expression_matrix_with_genome_set",
+        "differential_expression_matrix_no_genome_set",
+        "expression_set",
+        "feature_set_set",
+        "genome_set",
+        "reads_alignment_set",
+        "reads_set",
+    ]
+
+    # Randomly select a fixture name
+    selected_fixture = random.choice(fixtures)  # noqa: S311
+
+    # Use the request fixture to dynamically get the selected fixture
+    return request.getfixturevalue(selected_fixture)
```

Comment:
Now you can go back to [AssemblySet_basic_test](https://github.com/kbaseapps/SetAPI/pull/82/files#diff-36bd5310e554a2246108b4db27cd48a85df7882454dd2b9d139a5ffe0a6dc13a) with a complete understanding of what's going on!